addons/crm/models/crm_lead.py

```diff
@@ -1561,14 +1561,14 @@ def _merge_followers(self, opportunities):
     def _merge_log_summary(self, merged_followers, opportunities_tail):
         """Log the merge message on the lead."""
         self.ensure_one()
-        self.message_post_with_view(
+        self.message_post_with_source(
             "crm.crm_lead_merge_summary",
-            values={
+            render_values={
                 "merged_followers": merged_followers,
                 "opportunities": opportunities_tail,
                 "is_html_empty": is_html_empty,
             },
-            subtype_id=self.env.ref('mail.mt_note').id,
+            subtype_xmlid='mail.mt_note',
```

Comment:
Actually this commit adds support of subtype_xml for message_post_with_source, which was not the case previously. As it was decided during one of validation round, I added it in this commit because, well, enough commits. It could have been done previously, but the rebase would be a nightmare. Would say tant pis, in some modules some lines are modified twice but that's life.